### PR TITLE
[FIX] validate condition names early to avoid downstream error

### LINKED ISSUE #973: [ENH] Add in the doc how to not name a condition or a contrast to make bidsResults work
### Is there an existing issue for this?

- [X] I have searched the existing issues

### New feature

_No response_

### Unclear documentation

I have in my event file for example `condition_1` and `condition_2` and contrast `conditions` and `cond_1_gt_cond_2`.

When running the results workflow, it fails since the code is set to:

line 387 of `bidsResults.m`

``` 
% skip contrast with name ending in _[0-9]* as they are run level
% contrasts
```

So, I might suggest adding somewhere (FAQ? A warning within bidspm?) that naming a contrast `*_a-number` is not a good idea.

The behavior now is either 

1. in case it is asked to print only 1 contrast, the process does not start cause matlabbatch is empty, or 

2. in case of multiple contrasts where one is named "correctly" and the other is not it just thrown an error that `field dir does not exist` (if you want to know more about this error LMK cause it is a bit long to write here and not sure it is relevant)

ps: I lied, I did not check if an issue exists already OR if it is already written somewhere 😇

### PR DESCRIPTION
**Pull request by Remi-Gau**
fixes #973 

- validate condition names used on contrasts and dummy contrasts: they should not end with an underscore followed by one or more digits
- add FAQ entry: https://bidspm--983.org.readthedocs.build/en/983/FAQ.html#statistics-how-should-i-name-my-conditions-in-my-events-tsv
- link validation error message to FAQ entry

```
[11:58:36] bidspm - ERROR				BidsModel

For BIDS stats model named: "vislocalizer"
DummyConstrasts of Node run_level contains conditions ending with _[0-9]*: 'bar_2'.
This will lead to an error during results computation.
Change the name of your conditions.
See the FAQ: https://bidspm.readthedocs.io/en/latest/FAQ.html#statistics-how-should-i-name-my-conditions-in-my-events-tsv
```

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@marcobarilari 

Models will now be validated to avoid to have conditions names in contrast that could cause trouble.

Can you have a look at the new FAQ entry and at the error message and tell me if they are clear enough.

**Comment by Remi-Gau** [comment]
@marcobarilari 
OK I have changed the error to a warning because some demos did indeed have conditions ending in _1 or _2 and there was no error, so I prefer to not add extra constraints unless the problem is systematic.

File: docs/questions/statistics_naming-conditions.question.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+`happy_face1` and `house123` are ok, but `happy_face_1` and `house_123`  are not.
+
+If your BIDS dataset has conditions that do not follow this rule,
+then you can use
+the [`Replace` variable transform](https://github.com/bids-standard/variable-transform/blob/main/spec/munge.md#replace )
+in your [BIDS statistical model](https://bidspm.readthedocs.io/en/latest/bids_stats_model.html#transformation')
+to rename them on the fly without having to manually edit potentially dozens of files.
+
+See also example below.
```
Comment (marcobarilari) [inline docs/questions/statistics_naming-conditions.question.md:34]: AWESOME!!!!!!

File: demos/validate_models.m [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+% Validate all models in the demo folder
+%
+
+this_dir = fileparts(mfilename('fullpath'));
+
+addpath(fullfile(this_dir, '..'));
+
+bidspm();
+
+dirs = spm_select('FPList', this_dir, 'dir');
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply demos/validate_models.m:22]: added some comment and also a few more explanations at the top of the script

**Review by marcobarilari — APPROVED** [review]

**Comment by marcobarilari** [comment]
LGTM!

**Comment by Remi-Gau** [comment]
thanks
let's wait for the CI to finish and merge

**Pull request merged**